internal/provider/aws/provider.go

```diff
@@ -91,6 +91,28 @@ func WithNodeName(nodeName string) NodeNameFilter {
 	return NodeNameFilter{NodeName: nodeName}
 }
 
+func isFirewallRuleDuplicate(
+	firewallRules []v1alpha1.FirewallRule,
+	firewallRule *v1alpha1.FirewallRule,
+) bool {
+	firewallRules = slices.DeleteFunc(firewallRules, func(fr v1alpha1.FirewallRule) bool {
+		return fr.Name == firewallRule.Name
+	})
```

Comment:
The issue with that is that you're modifying the content of firewallRules, it will affect uses: 
https://go.dev/play/p/I4VJ7ADLmTO

I think instead maybe we should replace provider.ContainsPermission with slices.Contains, somethink like: 
```
	if frSpec.Direction == provider.DirectionIngress &&
		slices.Contains(provider.GetIngressIPPermissions(frSpecs), func(e ipPermission) {
			return fr.Name != firewallRule.Name && e.Equal(frSpec.IPPermission)
		})
		return true
	} else if frSpec.Direction == provider.DirectionEgress &&
		slices.Contains(provider.GetEgressIPPermissions(frSpecs), func(e ipPermission) {
			return fr.Name != firewallRule.Name && e.Equal(frSpec.IPPermission)
		})
		return true
	}
```